tools/PI/DevHome.PI/BarWindowHorizontal.xaml

```diff
@@ -77,38 +77,33 @@
                 <TextBlock Text="&#xe8a1;"/>
             </Button>
 
-            <Button
+            <controls:ExternalToolsManagementButton
                 x:Name="ManageExternalToolsButton"
                 x:Uid="ManageExternalToolsButton"
                 HorizontalAlignment="Center"
-                Click="ManageExternalToolsButton_Click">
-                <Button.ContextFlyout>
-                    <MenuFlyout x:Name="ExternalToolsMenu">
-                        <!-- We can't databind to MenuFlyout, so we'll add a MenuFlyoutItem
-                        for each external tool in codebehind here. -->
-                    </MenuFlyout>
-                </Button.ContextFlyout>
+                Click="{x:Bind _viewModel.ManageExternalToolsButton_Click}"
```

Comment:
For MVVM I suggest using commands instead of event handlers. There are several benefits for commands including decoupling the UI classes from the view-model. For example, you won't need to depend on `RoutedEventArgs` from the view-model and instead you can customize your own params. Commands are also reusable by multiple controls (again since we control the parameter and don't depend on a particular signature of a specific control). Now more specifically, the RelayCommand attribute from the MVVM toolkit it also allows controlling execution concurrency and has the option to bind the `CanExecute` to enable/disable a control.

TLDR; For UI manipulation (e.g. a button click that expands/collapses another control) I would use a regular event handler bound to a code-behind method. For everything else, I'd consider adding a `RelayCommand` in the view-model.

Related links:
- [RelayCommand attribute](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/relaycommand)

Reply:
I moved this particular instance to command binding. Ideally I would have also done it for the one below (_viewModel.ExternalToolButton_Click") but things get funky when you're in a datatemplate and it's not super straightward. I'll save moving that one for a rainy day.